kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/AccountManagerImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.kin.ecosystem;
+
+import android.support.annotation.NonNull;
+import com.kin.ecosystem.CreateTrustLineCall.TrustlineCallback;
+import com.kin.ecosystem.base.ObservableData;
+import com.kin.ecosystem.base.Observer;
+import com.kin.ecosystem.bi.EventLogger;
+import com.kin.ecosystem.bi.events.StellarAccountCreationRequested;
+import com.kin.ecosystem.bi.events.StellarKinTrustlineSetupFailed;
+import com.kin.ecosystem.bi.events.StellarKinTrustlineSetupSucceeded;
+import com.kin.ecosystem.bi.events.WalletCreationSucceeded;
+import com.kin.ecosystem.data.KinCallbackAdapter;
+import com.kin.ecosystem.data.auth.AuthDataSource;
+import com.kin.ecosystem.network.model.AuthToken;
+import kin.core.EventListener;
+import kin.core.KinAccount;
+import kin.core.ListenerRegistration;
+import kin.core.exception.OperationFailedException;
+
+class AccountManagerImpl implements AccountManager {
+
+	private static final String TAG = AccountManagerImpl.class.getSimpleName();
+
+	private final AccountManager.Local local;
+	private final EventLogger eventLogger;
+	private AuthDataSource authRepository;
+	private KinAccount kinAccount;
+	private final ObservableData<Integer> accountState;
+
+	private ListenerRegistration accountCreationRegistration;
+
+	AccountManagerImpl(@NonNull final AccountManager.Local local, @NonNull final EventLogger eventLogger) {
+		this.local = local;
+		this.eventLogger = eventLogger;
+		this.accountState = ObservableData.create(NONE);
+	}
+
+	@Override
+	public @AccountState
+	int getAccountState() {
+		return local.getAccountState();
+	}
+
+	@Override
+	public void addAccountStateObserver(@NonNull Observer<Integer> observer) {
+		accountState.addObserver(observer);
+	}
+
+	@Override
+	public void removeAccountStateObserver(@NonNull Observer<Integer> observer) {
+		accountState.removeObserver(observer);
+	}
+
+
+	void start(@NonNull final AuthDataSource authRepository, @NonNull final KinAccount kinAccount) {
+		new Log().withTag(TAG).put("setAccountState", "start").log();
+		this.authRepository = authRepository;
+		this.kinAccount = kinAccount;
+		this.setAccountState(local.getAccountState());
```

Comment:
you can add a check that if local.getAccountState() already equal CREATION_COMPLETED no need to call this.setAccountState , we might need also this.accountState.postValue(accountState)